pay-api/src/pay_api/services/cfs_service.py

```diff
@@ -378,40 +379,53 @@ def _build_lines(cls, payment_line_items: List[PaymentLineItemModel], negate: bo
-                account=f'{distribution_code.client}.{distribution_code.responsibility_centre}.'
-                        f'{distribution_code.service_line}.{distribution_code.stob}.'
-                        f'{distribution_code.project_code}.000000.0000'
-            )] if distribution_code else None
-            lines.append(
-                {
+
+            # Check if a line with same GL code exists, if YES just add up the amount. if NO, create a new line.
+            line = lines_map[distribution_code.distribution_code_id]
+
+            if not line:
+                index = index + 1
+                distribution = [dict(
+                    dist_line_number=index,
+                    amount=cls._get_amount(line_item.total, negate),
+                    account=f'{distribution_code.client}.{distribution_code.responsibility_centre}.'
+                            f'{distribution_code.service_line}.{distribution_code.stob}.'
+                            f'{distribution_code.project_code}.000000.0000'
+                )] if distribution_code else None
+
+                line = {
                     'line_number': index,
                     'line_type': CFS_LINE_TYPE,
                     'description': line_item.description,
                     'unit_price': cls._get_amount(line_item.total, negate),
                     'quantity': 1,
-                    'attribute1': line_item.invoice_id,
-                    'attribute2': line_item.id,
+                    # 'attribute1': line_item.invoice_id,
+                    # 'attribute2': line_item.id,
                     'distribution': distribution
                 }
-            )
+            else:
+                # Add up the price and distribution
+                line['unit_price'] = line['unit_price'] + cls._get_amount(line_item.total, negate)
+                line['distribution'][0]['amount'] = line['distribution'][0]['amount'] + \
+                    cls._get_amount(line_item.total, negate)
+
+            lines_map[distribution_code.distribution_code_id] = line
 
             if line_item.service_fees > 0:
                 service_fee_distribution: DistributionCodeModel = DistributionCodeModel.find_by_id(
                     distribution_code.service_fee_distribution_code_id)
-                index = index + 1
-                lines.append(
-                    {
+                service_line = lines_map[service_fee_distribution.distribution_code_id]
+
+                if not service_line:
+                    index = index + 1
+                    service_line = {
                         'line_number': index,
                         'line_type': CFS_LINE_TYPE,
                         'description': 'Service Fee',
                         'unit_price': cls._get_amount(line_item.service_fees, negate),
                         'quantity': 1,
-                        'attribute1': line_item.invoice_id,
-                        'attribute2': line_item.id,
```

Comment:
is both of them same?

Reply:
We no longer can use them as we are grouping for multiple lines as part of one line in CFS